### PR TITLE
Resjson support

### PR DESCRIPTION
**Pull request by MrMeison**
- Add async generation
- Add .resjson support
- Add vscode debug point
- Add nodejs 6 as dependencies
- Change CI configs for NodeJS 6

### CODE REVIEW

File: lib/VisualBuilder.js [hunk 1]
```diff
@@ -181,115 +188,312 @@ class VisualBuilder extends EventEmitter {
...
-            }
+            });
+    }
 
-            fs.writeFileSync(path.join(dropPath, 'pbiviz.json'), JSON.stringify(distPbiviz));
-            resolve();
+    _getDependencies(visualConfig) {
+        return new Promise((resolve, reject) => {
+            //load and validate dependencies if the file exists for the visual
+            if (!visualConfig.dependencies) return resolve();
```
Comment (ignatvilesov) [inline lib/VisualBuilder.js]: Please use curly brackets around of any ```if``` construction.

File: lib/VisualBuilder.js [hunk 2]
```diff
@@ -462,10 +665,10 @@ class VisualBuilder extends EventEmitter {
                 name: 'RScript',
                 handler: () => new Promise((resolve) => resolve())
             },
-            'stringResources/*.json': {
+            'stringResources/**/*.resjson': {
```
Comment (ignatvilesov) [inline lib/VisualBuilder.js]: Do we have to support ```.json``` as well?

File: lib/VisualBuilder.js [hunk 3]
```diff
@@ -665,7 +667,7 @@ class VisualBuilder extends EventEmitter {
                 name: 'RScript',
                 handler: () => new Promise((resolve) => resolve())
             },
-            'stringResources/**/*.resjson': {
+            '.resjson': {
```
Comment (ignatvilesov) [inline lib/VisualBuilder.js]: What about ```.json``` files? As far as I understand, it breaks back compatibility with the previous version of PBIVIZ. Is that correct?

File: lib/VisualBuilder.js [hunk 4]
```diff
@@ -181,115 +188,314 @@ class VisualBuilder extends EventEmitter {
...
-            resolve();
+    _getDependencies(visualConfig) {
+        return new Promise((resolve, reject) => {
+            //load and validate dependencies if the file exists for the visual
+            if (!visualConfig.dependencies) {
+                return resolve();
+            }
+            let dependenciesFilePath = this.visualPackage.buildPath(visualConfig.dependencies);
+            fs.access(dependenciesFilePath, (err) => {
+                if (err) return resolve();
```
Comment (ignatvilesov) [inline lib/VisualBuilder.js]: Please use curly brackets around the if statement.

File: package.json [hunk 5]
```diff
@@ -44,11 +44,15 @@
   "devDependencies": {
     "async": "^2.0.1",
```
Comment (ignatvilesov) [inline package.json]: Is it possible to get rid of ```^``` sign to install persistent versions?

File: spec/e2e/pbivizPackageSpec.js [hunk 6]
```diff
@@ -281,8 +281,211 @@ describe("E2E - pbiviz package", () => {
...
+                        .then(() => writeJsonPromise('stringResources/ru-RU/resources.resjson', ResJsonRuLocalization)),
+                    writeJsonPromise('stringResources/ru-RU.json', resourceStringRuLocalization)
+                ]))
+            .then(() => readJsonPromise('pbiviz.json'))
+            .then((pbivizJson) => {
+                pbivizJson.stringResources = ["stringResources/ru-RU.json"];
+                return writeJsonPromise('pbiviz.json', pbivizJson);
+            })
+            .then(() => {
+                try {
```
Comment (ignatvilesov) [inline spec/e2e/pbivizPackageSpec.js]: Why do we need ```try, catch``` statement?

Comment (MrMeison) [inline spec/e2e/pbivizPackageSpec.js]: Fixed

Comment (ignatvilesov) [inline spec/e2e/pbivizPackageSpec.js]: Are you sure that it has been fixed? I'm still able to find ```try``` statement. Why do we need it here?

File: spec/e2e/pbivizPackageSpec.js [hunk 7]
```diff
@@ -281,8 +281,204 @@ describe("E2E - pbiviz package", () => {
...
+        mkDirPromise('stringResources')
+            .then(() =>
+                Promise.all([
+                    mkDirPromise('stringResources/en-US')
+                        .then(() => writeJsonPromise('stringResources/en-US/resources.resjson', ResJsonEngLocalization)),
+                    mkDirPromise('stringResources/ru-RU')
+                        .then(() => writeJsonPromise('stringResources/ru-RU/resources.resjson', ResJsonRuLocalization))
+                ]))
+            .then(() => {
+                try {
```
Comment (ignatvilesov) [inline spec/e2e/pbivizPackageSpec.js]: The same ```try``` statement. Why do we need it?

**Pull request merged**